python/cudaq/kernel/ast_bridge.py

```diff
@@ -231,26 +231,68 @@ def getIntegerAttr(self, type, value):
         """
         return IntegerAttr.get(type, value)
 
-    def getFloatType(self):
+    def getFloatType(self, width=64):
         """
-        Return an MLIR float type (double precision).
+        Return an MLIR float type (single or double precision).
         """
-        return F64Type.get()
+        return F64Type.get() if width == 64 else F32Type.get()
 
     def getFloatAttr(self, type, value):
         """
-        Return an MLIR float attribute (double precision).
+        Return an MLIR float attribute (single or double precision).
         """
         return FloatAttr.get(type, value)
 
-    def getConstantFloat(self, value):
+    def getConstantFloat(self, value, width=64):
         """
         Create a constant float operation and return its MLIR result Value.
-        Takes as input the concrete float value. 
+        Takes as input the concrete float value.
+        """
+        ty = self.getFloatType(width=width)
+        return self.getConstantFloatWithType(value, ty)
+
+    def getConstantFloatWithType(self, value, ty):
+        """
+        Create a constant float operation and return its MLIR result Value.
+        Takes as input the concrete float value.
         """
-        ty = self.getFloatType()
         return arith.ConstantOp(ty, self.getFloatAttr(ty, value)).result
 
+    def getComplexType(self, width=64):
+        """
+        Return an MLIR complex type (single or double precision).
+        """
+        return self.getComplexTypeWithElementType(
+            self.getFloatType(width=width))
```

Comment:
We have to be careful here (document it?) since MLIR's complex<64> is the same as the numpy complex128 as far as I understand it. (I found this below.)

Reply:
Added comments, thanks!